src/Log/LogService.php

```diff
@@ -0,0 +1,186 @@
+
+    }
+
+    public function getConfig($configName){
+    	$this->config = \Zend_Registry::get($configName);
+    }
+
+    public function getPath(){
+    	return $this->logFilePath;
+    }
+
+    public function setPath($logFileName, $logPath = null){
+    	if($logPath == null)
+    		$this->logFilePath  = $this->config->workspacePath . '/log/' . $logFileName;
+    	else
+    		$this->logFilePath = $logPath . $logFileName;
+    	return $this->logFilePath;
+    }
+
+    public function getDefaultLog(){
+        if (array_key_exists('default.log', self::$logger))
+            return $logger = self::$logger['default.log'];
+        else{
+            return $this->setDefaultLog('default.log');
+        }
+    }
+
+    public function setDefaultLog($logFileName){
+        $this->openLog('Zend_Config', $logFileName);
+        return $this->setLog();
+    }
+
+    public function setLog($logger = null){
+
+	    if($logger == null)
+            $logger = $this->createLog();
+
+        if ($this->checkLogLevel() == false) {
+            $logLevelName = 'INFO';
+            $logger->warn('Log level not configured, using \'' . $logLevelName . '\'.');
+        } else {
+            $logLevelName = $this->checkLogLevel();
+        }
+
+        $zendLogRefl = new \ReflectionClass('Zend_Log');
+
+        $invalidLogLevel = false;
+
+        $logLevel = $zendLogRefl->getConstant($logLevelName);
+
+        if (empty($logLevel)) {
+            $logLevel = Zend_Log::INFO;
+            $invalidLogLevel = true;
+        }
+
+        // filter log output
+        $priorityFilter = new \Zend_Log_Filter_Priority($logLevel);
+        \Zend_Registry::set('LOG_LEVEL', $logLevel);	
+        $logger->addFilter($priorityFilter);
+
+        if ($invalidLogLevel) {
+            $logger->err('Invalid log level \'' . $logLevelName .
+                '\' configured.');
+        }
+
+        self::$logger[$this->logFileName] = $logger;      //how to know the logName for key value loggers other than logFileName
+        return $logger;
+    }
+
+    public function getLog($logName){      
+        if (array_key_exists($logName, self::$logger))
+            return $logger = self::$logger[$logName];
+        else{
+        	try{
+            	$this->openLog('Zend_Config', $logName);
+            	return $this->setLog();
+        	} catch {
+        		throw new Exception("Cannot create new log", 1);
+        		
+        	}
+        }
+	}
+
+    public function checkLogLevel(){		
+    	return isset($this->config->logging->log->level) ? $this->config->logging->log->level : false;
+    }
+
+	protected function createLog(){
+
+	    if ($this->format == null)
+	        $this->setFormat();
+
+        $formatter = new \Zend_Log_Formatter_Simple($this->format);
+
+        $writer = new \Zend_Log_Writer_Stream($this->logFile);	
+        $writer->setFormatter($formatter);
+
+        $logger = new \Zend_Log($writer);
+        return $logger;
+	}
+
+	public function setFormat($format = null){
```

Comment:
This function should be named __setDefaultFormat__, shouldn't it?

We also need a corresponding __getDefaultFormat__ function that is used even inside this class.

I think there are two separate things, the format string and creating the formatter object. I think creating the object should be placed in a function. I think we can do a bit more clean up here.